tests/components/ps4/test_media_player.py

```diff
@@ -0,0 +1,525 @@
+    ATTR_MEDIA_TITLE: MOCK_TITLE_NAME}
+
+MOCK_GAMES_DATA_LOCKED = {
+    ATTR_LOCKED: True,
+    ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_GAME,
+    ATTR_MEDIA_IMAGE_URL: MOCK_TITLE_ART_URL,
+    ATTR_MEDIA_TITLE: MOCK_TITLE_NAME}
+
+MOCK_STATUS_PLAYING = {
+    'host-type': MOCK_HOST_TYPE,
+    'host-ip': MOCK_HOST,
+    'host-request-port': MOCK_TCP_PORT,
+    'host-id': MOCK_HOST_ID,
+    'host-name': MOCK_HOST_NAME,
+    'running-app-titleid': MOCK_TITLE_ID,
+    'running-app-name': MOCK_TITLE_NAME,
+    'status': MOCK_STATUS_ON,
+    'status_code': MOCK_ON_CODE,
+    'device-discovery-protocol-version': MOCK_DDP_VERSION,
+    'system-version': MOCK_HOST_VERSION}
+
+MOCK_STATUS_IDLE = {
+    'host-type': MOCK_HOST_TYPE,
+    'host-ip': MOCK_HOST,
+    'host-request-port': MOCK_TCP_PORT,
+    'host-id': MOCK_HOST_ID,
+    'host-name': MOCK_HOST_NAME,
+    'status': MOCK_STATUS_ON,
+    'status_code': MOCK_ON_CODE,
+    'device-discovery-protocol-version': MOCK_DDP_VERSION,
+    'system-version': MOCK_HOST_VERSION}
+
+MOCK_STATUS_STANDBY = {
+    'host-type': MOCK_HOST_TYPE,
+    'host-ip': MOCK_HOST,
+    'host-request-port': MOCK_TCP_PORT,
+    'host-id': MOCK_HOST_ID,
+    'host-name': MOCK_HOST_NAME,
+    'status': MOCK_STATUS_STANDBY,
+    'status_code': MOCK_STANDBY_CODE,
+    'device-discovery-protocol-version': MOCK_DDP_VERSION,
+    'system-version': MOCK_HOST_VERSION}
+
+MOCK_STATUS_OFF = None
+
+MOCK_DEVICE = {
+    CONF_HOST: MOCK_HOST,
+    CONF_NAME: MOCK_NAME,
+    CONF_REGION: MOCK_REGION
+}
+
+MOCK_ENTRY_ID = 'SomeID'
+
+MOCK_DEVICE_MODEL = 'PlayStation 4'
+
+MOCK_DATA = {
+    CONF_TOKEN: MOCK_CREDS,
+    'devices': [MOCK_DEVICE]
+}
+
+MOCK_CONFIG = MockConfigEntry(
+    domain=DOMAIN, data=MOCK_DATA, entry_id=MOCK_ENTRY_ID)
+
+MOCK_LOAD = 'homeassistant.components.ps4.media_player.load_games'
+MOCK_SAVE = 'homeassistant.components.ps4.save_json'
+
+
+async def setup_mock_component(hass, entry=None):
+    """Set up Mock Media Player."""
+    if entry is None:
+        mock_entry = MockConfigEntry(
+            domain=ps4.DOMAIN, data=MOCK_DATA, version=VERSION,
+            entry_id=MOCK_ENTRY_ID)
+    else:
+        mock_entry = entry
+
+    mock_entry.add_to_hass(hass)
+
+    # Don't use an actual file.
+    with patch(MOCK_LOAD, return_value={}),\
+            patch(MOCK_SAVE, side_effect=MagicMock()):
+        await async_setup_component(hass, DOMAIN, {DOMAIN: {}})
+    await hass.async_block_till_done()
+
+    mock_entities = hass.states.async_entity_ids()
+
+    mock_entity_id = mock_entities[0]
+
+    return mock_entity_id
+
+
+async def mock_parse_status(hass, mock_device, load_data):
+    """Parse entity status with patched load_json."""
+    # Don't use an actual file.
+    with patch(MOCK_LOAD, return_value=load_data),\
+            patch(MOCK_SAVE, side_effect=MagicMock()),\
+            patch('os.path.isfile', return_value=True):
+        mock_device._parse_status()  # noqa: pylint: disable=protected-access
```

Comment:
Don't call this entity method directly. Let the entity update mechanism call it. Trigger the entity update by moving time forward. Do that by patching time and firing a time changed event.

See eg:
https://github.com/home-assistant/home-assistant/blob/b230562c76ce8c3b5c724ba89ee3f7cfb55ce1e9/tests/components/manual/test_alarm_control_panel.py#L75-L107